ingestion/src/test/java/feast/ingestion/ImportJobTest.java

```diff
@@ -180,91 +185,332 @@ public void runPipeline_ShouldWriteToRedisCorrectlyGivenValidSpecAndFeatureRow()
-                String.format(
-                    "Actual Redis value cannot be parsed as FeatureRow, key: %s, value :%s",
-                    key, new String(actualByteValue, StandardCharsets.UTF_8)));
+          FeatureRow actualValue = getFeatureRowFromByte(key, actualByteValue);
+
+          // Ensure the retrieved FeatureRow is equal to the ingested FeatureRow.
+          Assert.assertEquals(expectedValue, actualValue);
+        });
+    redisClient.shutdown();
+  }
+
+  /*
+   If there are missing entity fields in an incoming feature row, the row should not be ingested.
+  */
+
+  @Test
+  public void runPipeline_ShouldWriteFeatureRowsWithMissingEntityFieldsToDeadLetterQueue()
+      throws IOException, InterruptedException {
+
+    List<FeatureRow> input = new ArrayList<>();
+
+    LOGGER.info("Generating test data ...");
+    IntStream.range(0, IMPORT_JOB_SAMPLE_FEATURE_ROW_SIZE - FAILED_JOB_SAMPLE_FEATURE_ROW_SIZE)
+        .forEach(
+            i -> {
+              FeatureRow randomRow = TestUtil.createFeatureRowWithDefaults(spec, defaultValues);
+              input.add(randomRow);
+            });
+    IntStream.range(0, FAILED_JOB_SAMPLE_FEATURE_ROW_SIZE)
+        .forEach(
+            i -> {
+              FeatureRow randomRow = TestUtil.createFeatureRowWithDefaults(spec, defaultValues);
+              List<Field> fields =
+                  randomRow.getFieldsList().stream()
+                      .filter(field -> !(field.getName().equals("entity_id_primary")))
+                      .collect(Collectors.toList());
+              randomRow = randomRow.toBuilder().clearFields().addAllFields(fields).build();
+              input.add(randomRow);
+            });
+
+    Map<RedisKey, FeatureRow> expected = getExpectedRows(input, featureSet.getSpec());
+    PipelineResult pipelineResult = startImportJob(options);
+    Assert.assertEquals(pipelineResult.getState(), State.RUNNING);
+    publishFeatureRows(input, pipelineResult);
+
+    LOGGER.info("Validating the actual values written to Redis ...");
+    StatefulRedisConnection<byte[], byte[]> connection = redisClient.connect(new ByteArrayCodec());
+    RedisCommands<byte[], byte[]> sync = connection.sync();
+    expected.forEach(
+        (key, expectedValue) -> {
+
+          // Ensure ingested key exists.
+          byte[] actualByteValue = getBytesFromKey(sync, key);
+          if (actualByteValue == null) {
+            Assert.fail("Missing key in Redis.");
           }
 
+          // Ensure value is a valid serialized FeatureRow object.
+          FeatureRow actualValue = getFeatureRowFromByte(key, actualByteValue);
+
           // Ensure the retrieved FeatureRow is equal to the ingested FeatureRow.
           Assert.assertEquals(expectedValue, actualValue);
         });
     redisClient.shutdown();
   }
+
+  // get the expected rows from the input, and default values.
+  private Map<RedisKey, FeatureRow> getExpectedRows(List<FeatureRow> input, FeatureSetSpec spec) {
```

Comment:
I think it's good to briefly mention in comments how the `spec` argument is used to get the expected rows.
Also the method comment mention `default` values but this `default` values is nowhere in this method.